clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/ConnectionState.java

```diff
@@ -144,7 +161,11 @@ private void reconnect() {
   }
 
   private void connect() throws ConnectionException {
-    clusterConnection = LeasedConnectionFactory.connect(clusterUri, connectionProperties);
+    if( clusterUri != null){
```

Comment:
It seems like all this "If the old state is here do this, otherwise do that" is hiding the fact that there's a type hierarchy that should exist here.

Reply:
Created a `ConnectionSource` object which contains this information.